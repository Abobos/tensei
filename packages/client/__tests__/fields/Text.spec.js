import React from 'react'
import { render, screen } from '@testing-library/react'
import TextField from '~/fields/Text'

const field = {
    attributes: {},
    component: 'TextField',
    creationRules: [],
    databaseField: 'title',
    defaultValue: '',
    description: '',
    fieldName: 'Text',
    inputName: 'title',
    isNullable: true,
    isRelationshipField: false,
    isSearchable: true,
    isSortable: true,
    isUnique: true,
    name: 'Title',
    rules: [('required', 'max:24')],
    showOnCreation: true,
    showOnDetail: true,
    showOnIndex: true,
    showOnUpdate: true,
    sqlDatabaseFieldType: 'string',
    updateRules: [],
}

const props = { field }

test('matches snapshot', () => {
    const { asFragment } = render(<TextField {...props} />)

    expect(asFragment).toMatchSnapshot()
})
