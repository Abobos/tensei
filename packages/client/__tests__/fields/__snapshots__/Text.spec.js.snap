// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Text field component 1`] = `[Function]`;

exports[`Test the field components Test the Select field component 1`] = `[Function]`;

exports[`Test the field components Test the Text field component 1`] = `[Function]`;

exports[`Test the field components Test the TextArea field component 1`] = `[Function]`;

exports[`matches snapshot 1`] = `[Function]`;
